Helm

koenvanderveen opened the pull request.

tthoraldson committed:
default to 0 consumers

tthoraldson committed:
fix zmq rebase issue

tthoraldson committed:
add checks for producer port and consumer port

tthoraldson committed:
fix two more unit tests

koenvanderveen committed:
-

koenvanderveen committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into helm

koenvanderveen committed:
fix merge

koenvanderveen committed:
-

teo-milea committed:
added shorter helm versions

teo-milea committed:
fixed non syft shorter helm code

teo-milea committed:
found bug with serialization for hashabledict

teo-milea committed:
working shorter helm notebook with syft batch

koenvanderveen committed:
clean up pr

koenvanderveen committed:
fix testing setup

koenvanderveen committed:
Merge branch 'dev' of github.com:OpenMined/PySyft into helm

koenvanderveen committed:
clean up code

koenvanderveen committed:
-

teo-milea committed:
added parallel downloading and processing

teo-milea committed:
added new ExecuteOncePerCombination

teo-milea committed:
added test notebook for parallel download for lines

koenvanderveen committed:
trying to make docker work

teo-milea committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into demo_improvements

teo-milea committed:
modified jobs printing

teo-milea committed:
added input policy state saving

teo-milea committed:
added domain.jobs

teo-milea committed:
fix lint

teo-milea committed:
added test notebook

koenvanderveen committed:
-

koenvanderveen committed:
init mongo action store

koenvanderveen committed:
docker

koenvanderveen committed:
restructure notebook files

koenvanderveen commented on packages/grid/backend/grid/start.sh:
```diff
@@ -1,6 +1,7 @@
 #! /usr/bin/env bash
 set -e
 
+pip install nltk
```
todo: this is not a great solution

yashgorana replied:
@khoaguin perhaps we can move this into data_science optional dependencies

koenvanderveen commented on packages/grid/seaweedfs/start.sh:
```diff
@@ -1,6 +1,6 @@
 #! /usr/bin/env bash
 
 sleep 30 &&
-echo "s3.configure -access_key ${S3_ROOT_USER} -secret_key ${S3_ROOT_PWD} -user iam -actions Read,Write,List,Tagging,Admin -apply" \
```
todo: cleanup

koenvanderveen commented on packages/hagrid/hagrid/orchestra.py:
```diff
@@ -259,6 +262,7 @@ def deploy_to_python(
                 host=host,
```
todo: also add queue options here

koenvanderveen replied:
Not sure how we exactly want to organize default setup for docker/k8s? Do we want dynamic consumer creation on as default? Or do we want to have a default set of consumers? Currently async execution is dependent on consumers, so if we have none we cannot execute async work.

yashgorana replied:
Yes based on what we had demo'd in our meeting - we will have a fixed set of default workers and spawn custom env workers once they are approved by the DO.

koenvanderveen commented on packages/syft/src/syft/client/client.py:
```diff
@@ -333,6 +333,13 @@ def get_node_metadata(self, credentials: SyftSigningKey) -> NodeMetadataJSON:
+        if host is not None:
+            return GridURL(host_or_ip=host, port=8333).with_path(path)
+        else:
+            return GridURL(port=8333).with_path(path)
```
@shubham3121, I know we talked about this. I feel like I understand it well enough to patch the problem for situation but I dont know if I might be causing problems elsewhere, could we again have a look at this?

yashgorana replied:
This should be `port=Settings.S3_PORT` instead of `port=8333`

koenvanderveen replied:
grid is not a package though

koenvanderveen commented on packages/syft/src/syft/gevent_patch.py:
```diff
@@ -37,6 +34,6 @@ def is_notebook() -> bool:
 
 jupyter_notebook = is_notebook()
 
```
so I had to remove this to get rid of threading problems. There is a much longer story here, but I suspect we have to dedicate maybe 1 or 2 weeks to come up with a proper multithreading/processing plan

koenvanderveen commented on packages/syft/src/syft/node/node.py:
```diff
@@ -252,7 +263,8 @@ def __init__(
                 id = UID()
             self.id = id
-        self.packages = get_venv_packages()
+        self.packages = ""
```
this was breaking so I disabled i. it looks it wasnt doing anything

koenvanderveen commented on packages/syft/src/syft/node/node.py:
```diff
@@ -748,17 +852,25 @@ def get_role_for_credentials(self, credentials: SyftVerifyKey) -> ServiceRole:
+        self,
+        api_call: Union[SyftAPICall, SignedSyftAPICall],
+        job_id: Optional[UID] = None,
+        check_call_location=True,
```
I added `check_call_location` here because when a consumer reads a message the message node_uid does not match the nodes id as it was initially sent to a different node. I was not completely sure how to do this properly as the message is signed

koenvanderveen commented on packages/syft/src/syft/service/code/user_code.py:
```diff
@@ -675,11 +689,17 @@ def process_code(
+        # call_args = ["domain"] + call_args
+        context.output["uses_domain"] = True
+    call_stmt_keywords = [ast.keyword(arg=i, value=[ast.Name(id=i)]) for i in call_args]
     call_stmt = ast.Assign(
```
this slightly changed because we (optionally) added `domain` as an argument to usercode. In general I think this code could be more clear if we refactor it a bit

koenvanderveen commented on packages/syft/src/syft/service/code/user_code.py:
```diff
@@ -847,24 +869,177 @@ class UserCodeExecutionResult(SyftObject):
+            def __init__(self):
+                pass
+
+            def launch_job(self, func: UserCode, **kwargs):
```
most of the juggling with new usercode and approval can be removed once @teo-milea finishes the UX for the new policy

koenvanderveen commented on packages/syft/src/syft/service/code/user_code.py:
```diff
@@ -847,24 +869,177 @@ class UserCodeExecutionResult(SyftObject):
+        _locals = locals()
+
+        user_code_service = context.node.get_service("usercodeservice")
+        for user_code in user_code_service.stash.get_all(context.credentials).ok():
```
this is necessary because in the outer syft function function we pass in the the batch function name as the first argument to launch_job, e.g.
`domain.launch_job(run_batch, .....)` so this function needs to be present in the scope

koenvanderveen commented on packages/syft/src/syft/service/code/user_code.py:
```diff
@@ -847,24 +869,177 @@ class UserCodeExecutionResult(SyftObject):
+                from ... import UID
+
+                # get reference to node (TODO)
+                node = context.node
```
this is obviously the dangerous thing here. I am not sure how easy it would be to get a reference to this node object from within the syft function, but once you do you can basically do anything server side

koenvanderveen commented on packages/syft/src/syft/service/code/user_code.py:
```diff
@@ -878,7 +1053,13 @@ def execute_byte_code(code_item: UserCode, kwargs: Dict[str, Any]) -> Any:
     except Exception as e:
+        # stdlib
+        import traceback
+
```
maybe check dev_mode here

koenvanderveen commented on packages/syft/src/syft/service/code/user_code_service.py:
```diff
@@ -269,12 +289,22 @@ def call(
+            import traceback
+
+            # import ipdb
+            # ipdb.set_trace()
```
check dev_mode

koenvanderveen commented on packages/syft/src/syft/service/queue/queue.py:
```diff
@@ -78,31 +90,67 @@ def handle_message(message: bytes):
-
-        item = QueueItem(
+            print("HAD AN ERROR WHILE HANDLING MESSAGE")
+        # print("result", type(result.message.data))
```
check dev mode

teo-milea committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into demo_improvements

koenvanderveen committed:
cleanup pr

teo-milea committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into demo_improvements

koenvanderveen committed:
refactor docker compose files

koenvanderveen commented on packages/grid/docker-compose.multinode.yml:
```diff
@@ -0,0 +1,460 @@
+version: "3.8"
+services:
+  # docker-host:
```
we might get rid of this entire file eventually, I used it as a way to start testing the multinode setup

koenvanderveen committed:
add docker command instructions

teo-milea committed:
added job time estimation

teo-milea committed:
fixed concurrent logs issue

teo-milea committed:
moved new policy

teo-milea committed:
revert accidental removal of ExactMatch

teo-milea committed:
removed new policy from init

teo-milea committed:
some clean up

teo-milea committed:
changes after feedback

teo-milea committed:
fix lint

koenvanderveen committed:
finish docker setup

koenvanderveen committed:
clean up some stuff

koenvanderveen committed:
merge

teo-milea committed:
Merge branch 'dev' of github.com:OpenMined/PySyft into helm

koenvanderveen committed:
merge dev

teo-milea committed:
added new service methods

teo-milea committed:
added new seaweed docker files

teo-milea committed:
added new notebook for testing ux

koenvanderveen committed:
Merge pull request #8187 from OpenMined/dataset_ux

[WIP] Dataset ux

koenvanderveen committed:
fix merge

teo-milea committed:
replaced shell2http with flask server

koenvanderveen committed:
docker compose for seaweed moungting

koenvanderveen committed:
fixing docker setup

madhavajay committed:
Updated docker compose for hagrid

- Added docker clone existing container
- Improvements in seaweedfs container / build
- Turned off tracing and added try except for telemetry import

koenvanderveen committed:
fix linting

koenvanderveen committed:
fix reprs, add buffer for downloading

IonesioJunior committed:
Fix list repr when something goes wrong one of the jobs

teo-milea committed:
added safer context

teo-milea committed:
fix lint

koenvanderveen committed:
fix spawning up workers

koenvanderveen committed:
fix nb

koenvanderveen committed:
merge

koenvanderveen committed:
update port nb

koenvanderveen committed:
Merge pull request #8196 from madhavajay/madhava/helm_improvements

HELM: Updated docker compose for hagrid

teo-milea committed:
added context for producers

teo-milea committed:
delay task consumption if not all inputs are ready

teo-milea committed:
added notebook for testing

tcp committed:
fix: fix rounding issue in eta_string

tcp committed:
refactor: use divmod in eta_string format_timedelta

tcp committed:
lint: tox lint

tcp committed:
test: some initial tests for eta_string

tcp committed:
test: parametrized eta_string

tcp committed:
test: add float seconds timedelta case

tcp committed:
test: cleanup job instantiation

tcp committed:
lint: tox lint test

tcp committed:
lint: fix filename

koenvanderveen committed:
split out sequential job part, fix linting

koenvanderveen committed:
merge

koenvanderveen committed:
implement file ux

koenvanderveen committed:
Merge pull request #8253 from OpenMined/syftfileux

Syftfileux

koenvanderveen committed:
also show progress when done

koenvanderveen committed:
Merge branch 'helm-fix-time-rounding' of github.com:OpenMined/PySyft into helm-fix-time-rounding

koenvanderveen committed:
Merge branch 'helm-fix-time-rounding' of github.com:OpenMined/PySyft into helm

koenvanderveen committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into helm

IonesioJunior committed:
Update SyftLog Object Version to also store stderr logs

IonesioJunior committed:
ADD new log service to allow Node ADMIN inspect error logs

IonesioJunior committed:
ADD exception stack trace (server side) in case of an Exception

IonesioJunior committed:
ADD new flags to Job.log, allowing client to also retrieve error logs

teo-milea committed:
added launch_job visitor

teo-milea committed:
added nested requests to SubmitUserCode and UserCode

teo-milea committed:
changed service api to take into account nested requests

teo-milea committed:
added nested approval

teo-milea committed:
removed log line

teo-milea committed:
remove input policy state update

teo-milea committed:
saved progress

koenvanderveen committed:
refactor datastructures

koenvanderveen committed:
Merge pull request #8268 from OpenMined/data_structures_refactor

Data structures refactor

koenvanderveen committed:
merge with adding stacktrace + breaking code

koenvanderveen committed:
update reprs, include code when throwing exceptions, merge helm, include function name

koenvanderveen committed:
Merge pull request #8224 from OpenMined/fix_broken_subjobs_list

Fix list repr when something goes wrong one of the jobs

koenvanderveen committed:
remove debug code

teo-milea committed:
improved visitor search for launch_job

teo-milea committed:
removed nested requests at submit time

teo-milea committed:
added usercode search by service_func_name

teo-milea committed:
changed logic for code request

teo-milea committed:
added nested request logic

teo-milea committed:
changed request service to use nested requests

teo-milea committed:
fix merge

teo-milea committed:
fixed zmq queue after refactoring

teo-milea committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into nested_requests

IonesioJunior committed:
ADD new job_pid and JobStatus.INTERRUPTED variables

IonesioJunior committed:
Move a piece of handle_message function to be executed via multiprocessing

IonesioJunior committed:
ADD new job_service.kill method

IonesioJunior committed:
ADD Job restart service

IonesioJunior committed:
ADD Log.restart service

teo-milea committed:
added new permissions to queue items

teo-milea committed:
added error return to linked obj update

teo-milea committed:
added new queue permissions to handle message

teo-milea committed:
restricted usercode access in execute_byte_code

teo-milea committed:
changed usercode execution to use new permission

teo-milea committed:
changed action service to use new permission

teo-milea committed:
fixed bug with nested_requests

teo-milea committed:
removed cascading approval of nested user_codes

teo-milea committed:
added nested requests in UserCodeStatusChange repr

teo-milea committed:
added codes and nested requests to request repr

koenvanderveen committed:
protocol updates

koenvanderveen committed:
merge dev

koenvanderveen committed:
write migrations

koenvanderveen committed:
rewrite some funcs

IonesioJunior committed:
ADD subjob check in restart method

shubham3121 committed:
remove deleted classes from protocol.json

shubham3121 committed:
- reset protocol version to dev
- add missing migrations for subclasses of BlobRetrieval
- stage protocol changes

koenvanderveen committed:
Merge pull request #8270 from OpenMined/helm-migrations

Helm migrations

teo-milea committed:
added syft info message for nested_requests

teo-milea committed:
added identation to nested_requests repr

teo-milea committed:
moved nested requests from UserCodeStatusChange to UserCode

koenvanderveen committed:
-

koenvanderveen committed:
add a test

koenvanderveen committed:
add notebook

shubham3121 committed:
fix/bug: fix iteration over data items in migrate data in action store
fix/bug: add migrate_to method to passthrough attrs in action object
fix/bug: fix passing version of klass during migration in action store
fix/bug: handle dict of action object in tranform context

koenvanderveen committed:
fix stop param

koenvanderveen committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into helm

koenvanderveen committed:
Merge branch 'helm-tests' of github.com:OpenMined/PySyft into helm

IonesioJunior committed:
Merge branch 'helm' into feature/add_job_kill

teo-milea committed:
added test for nested_requests

teo-milea committed:
finish nested_requests test

teo-milea committed:
modified usercode migration

shubham3121 committed:
add pydantic attrs to pass through attrs in Action Object to make to_dict work
- update tranform context to use to_dict on object instead of __dict__

teo-milea committed:
added result object initialization when creating the job

teo-milea committed:
added syft_resolved field in action object

teo-milea committed:
added peek function to action_service

teo-milea committed:
working job promise

koenvanderveen committed:
test fixes

teo-milea committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into job_composition

koenvanderveen committed:
another test

koenvanderveen committed:
test with allowed inputs for node identity

koenvanderveen committed:
fix test accept deny

koenvanderveen committed:
Merge branch 'dev' of github.com:OpenMined/PySyft into helm

koenvanderveen committed:
fix notebook tests api

IonesioJunior committed:
Remove job kill from job service

IonesioJunior committed:
Remove unused package

IonesioJunior committed:
ADD Monitor thread to monitor jobs

IonesioJunior committed:
ADD INTERRUPTED Enum into QueueItem status enums to match with JobStatus enum

IonesioJunior committed:
Merge remote-tracking branch 'origin/feature/add_job_kill' into feature/add_job_kill

IonesioJunior committed:
ADD psutil as a syft dependency

IonesioJunior committed:
Merge branch 'helm' into feature/job_restart

IonesioJunior committed:
Merge branch 'helm' into feature/add_job_kill

IonesioJunior committed:
Remove artificial JobItem from protocol_version

IonesioJunior committed:
Merge remote-tracking branch 'origin/feature/add_job_kill' into feature/add_job_kill

teo-milea committed:
added test notebook

teo-milea committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into nested_requests

teo-milea committed:
fixed usercodeV1

teo-milea committed:
usercodestatuschange version

teo-milea committed:
added queue versioning

teo-milea committed:
fixed object versioning

koenvanderveen committed:
fix notebooks tests except enclave tests

teo-milea committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into job_composition

teo-milea committed:
fix action None in node.py

koenvanderveen committed:
fix .get.get()

teo-milea committed:
fixed request test failling

teo-milea committed:
latest versions

koenvanderveen committed:
Merge branch 'job_composition' of github.com:OpenMined/PySyft into helm

teo-milea committed:
add test notebook

koenvanderveen committed:
lint

koenvanderveen committed:
merge

rasswanth-s commented on packages/grid/seaweedfs/new_seaweed.sh:
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+echo "got api key"
+echo ${STACK_API_KEY}
+export STACK_API_KEY=${STACK_API_KEY}
```
Q: Is the syft stack api key used in seaweedfs ?

yashgorana replied:
This looks the same as start.sh

rasswanth-s commented on packages/grid/seaweedfs/seaweedfs.dockerfile:
```diff
@@ -0,0 +1,18 @@
+RUN chmod +x /start.sh
+RUN chmod +x /mount_command.sh
+
+CMD ["bash", "/seaweed.sh"]
```
Q: I think seaweed.sh is not copied, for running the custom script?

yashgorana replied:
It should either be start.sh or new_seaweed.sh

yashgorana replied:
We must expose all environment variables with default values.

```
ENV SEAWEED_MOUNT_PORT=4001 \
    STACK_API_KEY=""
```

yashgorana commented on packages/grid/docker-compose.multinode.yml:
```diff
@@ -0,0 +1,461 @@
+    # depends_on:
+    #   - "docker-host"
+
+  tailscale:
```
@shubham3121 @koenvanderveen  we have removed tailscale on "dev". perhaps we have to sync this file against dev as well. Please let me know if we still need tailscale for multinode setup.

koenvanderveen replied:
this file can be removed probably

yashgorana commented on packages/grid/docker-compose.multinode.yml:
```diff
@@ -0,0 +1,461 @@
+    labels:
+      - "orgs.openmined.syft=this is a syft backend container"
+
+  backendworker3:
```
We might want to use replicas over here instead of different instances of backendworkers

koenvanderveen replied:
see other comment

yashgorana commented on packages/grid/default.env:
```diff
@@ -50,6 +50,9 @@ DOMAIN_CHECK_INTERVAL=60
 DEV_MODE=False
+QUEUE_PORT=5556
+CREATE_PRODUCER=False
+N_CONSUMERS=0
```
Env vars present in dockerfile but not here:
* ~`HOST_GRID_PATH`~ (set as pwd from dockerfile, and run only in dev mode, so perhapse set it to empty instead?)
* `BACKEND_STORAGE_PATH`
* `SEAWEED_MOUNT_PORT`
* `STACK_API_KEY`

yashgorana commented on packages/grid/seaweedfs/app.py:
```diff
@@ -0,0 +1,34 @@
+
+
+@app.route("/configure_azure", methods=["POST"])
+def test() -> str:
```
rename this function to configure_azure?

yashgorana commented on packages/grid/seaweedfs/app.py:
```diff
@@ -0,0 +1,34 @@
+
+@app.route("/configure_azure", methods=["POST"])
+def test() -> str:
+    first_res = json.loads(request.data.decode("utf-8").replace("'", '"'))
```
we need to manager errors here, because we may pass incorrect JSON payload, and the error needs to be returned accordingly.

yashgorana commented on packages/grid/seaweedfs/app.py:
```diff
@@ -0,0 +1,34 @@
+    remote_name = first_res["remote_name"]
+    bucket_name = first_res["bucket_name"]
+
+    res = subprocess.run(
```
There might be a subprocess error here
* There was en error starting the command
* or, There was an error in mount_command.sh
* or, there was a system error resulting in failed mount.
These must  be returned.

yashgorana replied:
Good To have: Rate limit this api at an HTTP level or at the subprocess invocation to avoid spawning too many `bash mount_command.sh`

yashgorana commented on packages/grid/seaweedfs/mount_command.sh:
```diff
@@ -0,0 +1,15 @@
+
+
+
+echo "remote.configure -name=$1 -type=azure -azure.account_name=$2 -azure.account_key=$5" | weed shell
```
duplicate of line#3?

shubham3121 replied:
I think we can remove  this, might be for just debugging.

yashgorana commented on packages/grid/seaweedfs/mount_command.sh:
```diff
@@ -0,0 +1,15 @@
+echo "$1 $2 $3 $4 $5"
```
Perhaps we can make this a bit more descriptive like

```
echo remote_name=$1 account_name=$2...
```

teo-milea replied:
this was only for debugging, we could remove it all together

yashgorana commented on packages/grid/traefik/docker/dynamic.yml:
```diff
@@ -40,6 +48,22 @@ http:
+      service: "seaweedfsmount"
+      middlewares:
+        - "blob-storage-mount-url"
+    vpn:
```
vpn has been removed. cc: @shubham3121

yashgorana commented on packages/hagrid/hagrid/orchestra.py:
```diff
@@ -314,12 +322,17 @@ def deploy_to_python(
             raise NotImplementedError(f"node_type: {node_type_enum} is not supported")
+
+        def stop() -> None:
+            worker.stop()
```
check if worker is defined, because it is instantiated only when `node_type_enum in worker_classes`

koenvanderveen replied:
I think its always there (if you have no port)

yashgorana commented on packages/syft/src/syft/client/client.py:
```diff
@@ -153,7 +153,7 @@ def get_cache_key(self) -> str:
         return self.url.with_path(self.routes.ROUTE_API_CALL.value)
 
-    def to_blob_route(self, path: str) -> GridURL:
+    def to_blob_route(self, path: str, **kwargs) -> GridURL:
```
do we need this?

koenvanderveen replied:
I think so, but we would have to look more deeply into doing this properly. We are using it in BlobRetrievalByURL._read_data and  SeaweedFSBlobDeposit.write. Depending on the deployment settings we need both but there is probably a better way of solving this

yashgorana commented on packages/syft/src/syft/node/node.py:
```diff
@@ -334,16 +352,48 @@ def __init__(
+        if not self.dev_mode:
+            raise ValueError("You can only get host location in dev mode")
+        # We set this variable in docker compose (from pwd)
+        env_val = os.environ.get("HOST_GRID_PATH", None)
```
PWD can be anything inside syft. For example, while developing grid or building dockerfiles we need to be in` packages/` or `packages/grid`... so perhaps we can just look for tox.ini and keep traversing to the parent, that way we eliminate the need for any env var altogether.

yashgorana commented on packages/syft/src/syft/node/node.py:
```diff
@@ -794,13 +895,37 @@ def __eq__(self, other: Any) -> bool:
+            else:
+                res = result.ok()
+                if res.status == Status.COMPLETED:
+                    return res
```
need to check for errors if `res.status != Completed` then what?

shubham3121 replied:
Maybe we can check for status is Error/Failed and return

yashgorana reviewed (commented):
just adding a couple of comments for the team to get this PR ready for merge.

koenvanderveen committed:
fix

koenvanderveen committed:
merge job restart

koenvanderveen committed:
fix problems with worker serialization

koenvanderveen committed:
merge dev

shubham3121 committed:
add and set migrate flag only during node orchestration

shubham3121 committed:
Merge branch 'dev' into helm

koenvanderveen committed:
debug blob storage access via global scope

koenvanderveen committed:
reset locks when resetting node

koenvanderveen committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into feature/add_job_kill

koenvanderveen committed:
skip some tests on mac

koenvanderveen committed:
Merge pull request #8272 from OpenMined/feature/add_job_kill

ADD job.kill()

koenvanderveen committed:
fix bug with enclaves

yashgorana commented on packages/syft/src/syft/service/queue/queue.py:
```diff
@@ -38,8 +86,15 @@ def create_consumer(
-        return self._client.add_producer(queue_name=queue_name)
+    def create_message_queue(self, queue_name: str):
+        return self._client.add_message_queue(queue_name)
+
```
Don't see `add_message_queue` implemented and `create_message_queue` used anywhere. Should we remove this?

yashgorana commented on packages/syft/src/syft/service/queue/queue.py:
```diff
@@ -78,31 +224,41 @@ def handle_message(message: bytes):
+        job_item.node_uid = worker.id
+
+        worker.queue_stash.set_result(credentials, queue_item)
+        worker.job_stash.set_result(credentials, job_item)
```
Instead of duplicating this we can instead just forward the message bytes to the process function `handle_message_multiprocessing` where we can run this routine followed by handling and setting results.

```
@staticmethod
def handle_message(message: bytes):
    p = multiprocessing.Process(
            target=handle_message_multiprocessing,
            args=(message),
        )
    p.start()
    p.join()
```

yashgorana replied:
Aditionally, we can have fixed size process pool of size "n", so that we can process "n" messages without blocking the worker

yashgorana commented on packages/syft/src/syft/service/code/user_code.py:
```diff
@@ -862,24 +977,195 @@ class UserCodeExecutionResult(SyftObject):
+        if context.job is not None:
+            job_id = context.job_id
+            log_id = context.job.log_id
+
+            def print(*args, sep=" ", end="\n"):
+                def to_str(arg: Any) -> str:
+                    if isinstance(arg, bytes):
+                        return arg.decode("utf-8")
+                    if isinstance(arg, Job):
+                        return f"JOB: {arg.id}"
+                    if isinstance(arg, SyftError):
+                        return f"JOB: {arg.message}"
+                    if isinstance(arg, ActionObject):
+                        return str(arg.syft_action_data)
+                    return str(arg)
+
+                new_args = [to_str(arg) for arg in args]
+                new_str = sep.join(new_args) + end
+                log_service = context.node.get_service("LogService")
+                log_service.append(context=context, uid=log_id, new_str=new_str)
+                time = datetime.datetime.now().strftime("%d/%m/%y %H:%M:%S")
+                return __builtin__.print(
+                    f"{time} FUNCTION LOG ({job_id}):",
+                    *new_args,
+                    end=end,
+                    sep=sep,
+                    file=sys.stderr,
+                )
+
```
A few key points:
- This is only capturing logs generated from print function & not logging std lib.
- For each print() we are triggering a Mongo API call - any failure exception in this API or downstream monog server outage results in direct failure of the function and the job even though the function itself has no issues.
- There's no rate limit on the add/apend API call itself hence a `while(True): print(...)` can probably overwhelm the single-node mongo server.

I would recommend that we setup logging & patch print to write to a file in /var/log instead. Once the function completes (or errors out) we simply upload the logs in a single Mongo API call.

koenvanderveen replied:
yes those are definitely next steps. Not sure about writing the logs only after finishing. The whole point of logs is that you can see them as the program runs. This is especially useful if its a long running job

yashgorana commented on packages/syft/src/syft/service/code/user_code.py:
```diff
@@ -862,24 +977,195 @@ class UserCodeExecutionResult(SyftObject):
+            data_protcol = get_data_protocol()
+            user_api = node.get_api(context.credentials, data_protcol.latest_version)
+            user_api.signing_key = user_signing_key
+            # We hardcode a python connection here since we have access to the node
+            # TODO: this is not secure
+            user_api.connection = PythonConnection(node=node)
+
```
Must discuss this

koenvanderveen committed:
security fixes + add flaky for zmq test

koenvanderveen committed:
add flaky

koenvanderveen committed:
merge dev

koenvanderveen committed:
fix circular import

koenvanderveen committed:
mark syft function test as flaky

koenvanderveen committed:
fix imports

koenvanderveen committed:
ulimit

koenvanderveen committed:
tox

kiendang committed:
Detect json lines based on file extension

kiendang committed:
Fix typo

kiendang commented on packages/syft/src/syft/store/linked_obj.py:
```diff
@@ -65,6 +65,7 @@ def update_with_context(
         )
         if result.is_ok():
             return result
+        return result
```
This could just be `return result`.

kiendang committed:
Remove redundant is_ok check

koenvanderveen committed:
skip zmq tests on windows

koenvanderveen committed:
Merge branch 'helm' of github.com:OpenMined/PySyft into helm

kiendang reviewed (commented):
nitpick: for the notebooks under `notebooks/helm` could we number them so that they're easier to navigate? Also could we use kebab case for all of them like the other notebooks? Right now there are `dataset_ux.ipynb`, `consumer node.ipynb`, `helm-syft-mongo.ipynb`. There's one `Untitled.ipynb` as well.

koenvanderveen committed:
fixtures for zmq tests

koenvanderveen committed:
remove protocol files

koenvanderveen committed:
kill zmq properly on flaky retries

koenvanderveen committed:
-

koenvanderveen committed:
-

koenvanderveen committed:
-

khoaguin committed:
add some type annotations for queue consumer and producer

shubham3121 committed:
fix zmq_queue test
- fix incorrect reference of close method for producer/consumer
- fix closing of threads
- rename stop to close
- scope locking to Producer class
- linting
Co-authored-by: Kien Dang <mail@kien.ai>

shubham3121 committed:
add missing lock at consumer level

shubham3121 committed:
revert to creating a global lock

Co-authored-by: Kien Dang <mail@kien.ai>

shubham3121 committed:
remove locking from sqlite backing store

Co-authored-by: Koen van der Veen <koenlennartvanderveen@gmail.com>

koenvanderveen committed:
fix test

koenvanderveen committed:
fix fixture

koenvanderveen committed:
fix sec

IonesioJunior committed:
ADD missing env vars in default.env

IonesioJunior committed:
Update default.env

Remove API_KEY to  force people to set them in their own environments.

IonesioJunior committed:
Remove unused script / fix seaweedfs docker entrypoint

kiendang commented on packages/syft/src/syft/service/action/action_service.py:
```diff
@@ -104,6 +106,50 @@ def set(
+        uid: UID,
+        twin_mode: TwinMode = TwinMode.PRIVATE,
+    ) -> Result[Ok[ActionObject], Err[str]]:
+        """Get an object from the action store"""
+        return self._peek(context, uid, twin_mode)
+
+    def _peek(
+        self,
+        context: AuthedServiceContext,
+        uid: UID,
+        twin_mode: TwinMode = TwinMode.PRIVATE,
+        has_permission=False,
+    ) -> Result[Ok[ActionObject], Err[str]]:
+        """Get an object from the action store"""
+
+        result = self.store.get(
+            uid=uid, credentials=context.credentials, has_permission=has_permission
+        )
+
+        if result.is_ok():
+            obj = result.ok()
+            obj._set_obj_location_(
+                context.node.id,
+                context.credentials,
+            )
+            if isinstance(obj, TwinObject):
+                if twin_mode == TwinMode.PRIVATE:
+                    obj = obj.private.as_empty()
+                    obj.syft_point_to(context.node.id)
+                elif twin_mode == TwinMode.MOCK:
+                    obj = obj.mock.as_empty()
+                    obj.syft_point_to(context.node.id)
+                else:
+                    obj.mock.syft_point_to(context.node.id)
+                    obj.private.syft_point_to(context.node.id)
+            else:
+                obj = obj.as_empty()
+            return Ok(obj)
+        return result
+
```
Are these only for debugging purposes and should be removed?

kiendang commented on packages/grid/seaweedfs/mount_command.sh:
```diff
@@ -0,0 +1,15 @@
+        -azure.account_key=$5" \
+        | weed shell
+
+echo 's3.bucket.create -name $3 ' | weed shell
```
This should use double quotes? Otherwise `$3` will not be substituted. Try
```sh
HELLO="hello"
echo '$HELLO'
echo "$HELLO"
```

shubham3121 committed:
Merge pull request #8298 from OpenMined/hotfix/hb_880

Remove unused script / fix seaweedfs docker entrypoint

shubham3121 committed:
Merge branch 'helm' into hotfix/hb_881

shubham3121 committed:
Merge pull request #8297 from OpenMined/hotfix/hb_881

ADD missing env vars in default.env

madhavajay committed:
Fixed SQLiteBackingStore to handle concurrency

- Added SyftLock to prevent concurrent writes as SQLite doesn't like it
- Changed Connection re-use to prevent explosion of open files
- Changed destructor to only close connection on last __del__
- NOTE: This might be a bit slower, we can attempt to improve if it works
- Added scripts/print_fd.py debug tool for showing open file descriptors

co-author: @tcp

madhavajay committed:
Changed hardcoded /tmp path to tempfile.TemporaryDirectory

koenvanderveen committed:
Merge pull request #8300 from madhavajay/madhava/fix-sqlite-fd

Fix SQLite File Descriptor Leak

koenvanderveen committed:
-

koenvanderveen committed:
fix notebook test

koenvanderveen committed:
fix stack tests

koenvanderveen committed:
fix notebook

koenvanderveen committed:
fix few comments

koenvanderveen committed:
remove settings

koenvanderveen committed:
fix notebook

koenvanderveen committed:
trigger CI

koenvanderveen committed:
fix nb

koenvanderveen committed:
fix typo

koenvanderveen committed:
Merge branch 'helm' into file-debugging

koenvanderveen committed:
Merge pull request #8292 from OpenMined/file-debugging

dont merge: CI filedescriptor debugging

koenvanderveen committed:
clean up notebooks
